### PR TITLE
Update RUST MSRV and address clippy issues.

### PR DESCRIPTION
**Pull request by d0cd**
This PR updates the Rust MSRV to 1.81.0 and fixes outstanding Clippy issues.

### CODE REVIEW

File: synthesizer/program/src/logic/instruction/operation/macros.rs [hunk 1]
```diff
@@ -775,7 +775,9 @@ mod tests {
                                 }
                                 // These indicators are later used in the for-loops below.
                                 is_shift_operator |= true;
-                                shift_exceeds_bitwidth |= ((*b as u32) >= $input_a::<CurrentNetwork>::size_in_bits() as u32);
+                                let input_a_size_in_bits = u32::try_from($input_a::<CurrentNetwork>::size_in_bits()).expect("Input size in bits exceeded u32::MAX");
+                                shift_exceeds_bitwidth |= ((*b as u32) >= input_a_size_in_bits);
+                                shift_exceeds_bitwidth |= ((*b as u32) >= input_a_size_in_bits);
```
Comment (vicsn) [inline synthesizer/program/src/logic/instruction/operation/macros.rs:780]: Was this line meant to be duplicated?

**Review by vicsn — APPROVED** [review]
LGTM so far except for the one comment

File: synthesizer/program/src/logic/instruction/operation/macros.rs (hunk 1 above)
Comment (d0cd) [reply synthesizer/program/src/logic/instruction/operation/macros.rs:780]: Good catch!

**Comment by vicsn** [comment]
One more thing, it's also worth replacing `cache_key: v1` to `cache_key: v1.81` in the `.circleci/config.yml`. The new cache key will ensure new 1.81 .cargo dependencies will be cached. And by giving it the same name as the MSRV, hopefully it won't be overlooked in the future.

The snarkOS circleci config doesn't have a similar cache key setup yet because it was not a priority yet.

**Comment by d0cd** [comment]
Closing in favor of https://github.com/AleoNet/snarkVM/pull/2558

**Pull request closed without merging**